File: apps/frontend/src/types/llm.ts
export interface LLMInvocation {
  id: string;
  workflow_id?: string | null;
  agent_execution_id?: string | null;
  agent_name?: string | null;
  provider: string;
  model: string;
  mode: string;
  prompt_hash?: string | null;
  input_tokens?: number | null;
  output_tokens?: number | null;
  total_tokens?: number | null;
  status: string;
  error_message?: string | null;
  latency_ms?: number | null;
  fallback_used: boolean;
  structured_output_valid: boolean;
  created_at: string;
}

export interface LLMConfig {
  live_ai_enabled: boolean;
  api_key_present: boolean;
  provider: string;
  model: string;
  reasoning_model: string;
  structured_outputs_enabled: boolean;
  fallback_to_demo: boolean;
  cache_enabled: boolean;
  prompt_logging_enabled: boolean;
  response_logging_enabled: boolean;
}

export interface LLMTestResponse {
  enabled: boolean;
  fallback_used: boolean;
  status: string;
  provider: string;
  model: string;
  message: string;
  latency_ms?: number | null;
}

File: apps/backend/app/schemas/llm.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class LLMInvocationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    workflow_id: UUID | None = None
    agent_execution_id: UUID | None = None
    agent_name: str | None = None
    provider: str
    model: str
    mode: str
    prompt_hash: str | None = None
    input_tokens: int | None = None
    output_tokens: int | None = None
    total_tokens: int | None = None
    status: str
    error_message: str | None = None
    latency_ms: int | None = None
    fallback_used: bool
    structured_output_valid: bool
    created_at: datetime


class LLMConfigRead(BaseModel):
    live_ai_enabled: bool
    api_key_present: bool
    api_key_usable: bool | None = None
    api_key_state: str | None = None
    provider: str
    model: str
    reasoning_model: str
    structured_outputs_enabled: bool
    fallback_to_demo: bool
    cache_enabled: bool
    prompt_logging_enabled: bool
    response_logging_enabled: bool


class LLMTestResponse(BaseModel):
    success: bool | None = None
    enabled: bool
    fallback_used: bool
    status: str
    provider: str
    model: str
    message: str
    error_message: str | None = None
    latency_ms: int | None = None
